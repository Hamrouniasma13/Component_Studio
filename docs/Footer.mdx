---
name: Footer
route: /footer
menu: 🌐 Organisms
---

import { Playground } from 'docz'
import { Footer } from '../src'

# Footer

<Playground>
<Footer
  link={{ label: 'Mentions légales', url: '/' }}
  socials={[
    {
      icon: 'cog',
      url: 'http://wwww.twitter.com',
    },
    {
      icon: 'cog',
      url: 'http://wwww.twitter.com',
    },
    {
      icon: 'cog',
      url: 'http://wwww.twitter.com',
    },
    {
      icon: 'cog',
      url: 'http://wwww.twitter.com',
    },
  ]}
  copyright={'© 2019 AXA All Rights Reserved'}
/>

</Playground>
