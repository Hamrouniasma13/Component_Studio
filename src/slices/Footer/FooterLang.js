/** @jsx jsx */
import { jsx } from '@emotion/core'
import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'
import { useClickAway } from 'react-use'
import Icon from '../../components/Icon'
import AnimatedIcon from '../../components/AnimatedIcons'
import { lighten } from '../../theme/utils'

const LangLabel = styled('div')(({ theme }) => ({
  color: theme.colors.darkGray,
  textTransform: 'capitalize',
}))

const LangList = styled('div')(({ theme }) => ({
  border: `1px solid ${theme.colors.gray}`,
  marginTop: '10px',
}))

const LangListItem = styled('div')(({ theme }) => ({
  listStyle: 'none',
  padding: '15px',
  color: theme.colors.darkGray,
  fontWeight: 600,
  borderBottom: `1px solid ${theme.colors.gray}`,
  backgroundColor: 'white',
  transition: 'background-color 0.2s',
  '&:hover': {
    backgroundColor: theme.colors.lightGray,
  },
  '&:last-child': {
    borderBottom: '0',
  },
}))

const LangListItemSelected = styled('div')(({ theme, selected }) => ({
  color: selected ? theme.colors.primary : 'inherit',
}))

const FooterLangList = styled('div')(({ theme, isOpen }) => ({
  background: 'white',
  padding: '20px',
  color: theme.colors.gray,
  userSelect: 'none',
  display: isOpen ? 'block' : 'none',
  [theme.media.tablet]: {
    boxShadow: 'rgba(0,0,0,0.3) 1px 1px 15px 0px',
    position: 'absolute',
    bottom: '100%',
    left: 0,
    right: 0,
    display: 'block',
    transition: isOpen
      ? 'transform 0.3s, opacity 0.1s ease 0.1s'
      : 'transform 0.1s ease 0.1s, opacity 0.3s',
    opacity: isOpen ? 1 : 0,
    transform: isOpen ? 'translateY(0)' : 'translateY(5px)',
    pointerEvents: isOpen ? 'auto' : 'none',
  },
}))

function LangSwitcherList({ lang, langs, title, onLangChange, isOpen }) {
  return (
    <FooterLangList isOpen={isOpen}>
      <LangLabel>{title}</LangLabel>
      <LangList>
        {langs.map((_lang, i) => (
          <LangListItem key={i} onClick={() => onLangChange(_lang.id)}>
            <LangListItemSelected selected={_lang.id === lang.id}>
              {_lang.label}
            </LangListItemSelected>
          </LangListItem>
        ))}
      </LangList>
    </FooterLangList>
  )
}

const FooterLangInner = styled('div')(({ theme }) => ({
  label: 'FooterLang',
  borderBottom: `1px solid ${lighten(theme.colors.primary, 15)}`,
  textTransform: 'uppercase',
  fontSize: '14px',
  letterSpacing: '.6px',
  cursor: 'pointer',

  [theme.media.tablet]: {
    minWidth: '320px',
    margin: 0,
    borderBottom: 0,
    borderRight: `1px solid ${lighten(theme.colors.primary, 15)}`,
    transition: 'background-color 0.2s',
    position: 'relative',
    left: '-15px',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.3)',
    },
  },
}))

const SelectedLangWrapper = styled('div')({
  position: 'relative',
})

const SelectedLang = styled('div')({
  paddingLeft: '15px',
  height: '55px',
  display: 'flex',
  alignItems: 'center',
})

const StyledAnimatedIcon = styled(AnimatedIcon.chevron)(({ size = 20 }) => ({
  position: 'absolute',
  top: '50%',
  right: '15px',
  transformOrigin: 'center',
  marginTop: -size / 2,
}))

export default function FooterLang(props) {
  const [open, setOpen] = React.useState(false)
  const ref = React.useRef(null)
  useClickAway(ref, () => setOpen(false))

  return (
    <FooterLangInner ref={ref}>
      <SelectedLangWrapper onClick={() => setOpen(!open)}>
        <SelectedLang>
          <Icon
            css={{ marginRight: '10px', verticalAlign: 'sub' }}
            name="globe"
            size={20}
          />
          <span>{props.lang.label || 'Fran√ßais'}</span>
        </SelectedLang>
        <StyledAnimatedIcon size={20} open={open} />
      </SelectedLangWrapper>
      <LangSwitcherList {...props} isOpen={open} />
    </FooterLangInner>
  )
}

FooterLang.propTypes = {
  lang: PropTypes.shape({
    ID: PropTypes.string,
    label: PropTypes.string,
  }).isRequired,
  langs: PropTypes.arrayOf(
    PropTypes.shape({
      ID: PropTypes.string,
      label: PropTypes.string,
    })
  ).isRequired,
  title: PropTypes.string,
  onLangChange: PropTypes.func.isRequired,
}
