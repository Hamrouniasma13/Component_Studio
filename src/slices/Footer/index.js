import React from 'react'
import PropTypes from 'prop-types'

/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'
import useMedia from 'react-use/lib/useMedia'

import typos from '../../typos'

import Icon from '../../components/Icon'
import Link from '../../components/Link'
import Container from '../../components/Container'

const Footer = ({ link, socials, copyright, className }) => {
  const { colors, breakpoints, media } = React.useContext(ThemeContext)

  const isMobile = useMedia(`(max-width: ${breakpoints.tablet}px)`)

  const styles = {
    footer: {
      padding: '22px 0',
      height: 85,
      width: '100%',
      backgroundColor: colors.darkerGray,
      color: colors.lighterGray,
      [media.tablet]: {
        padding: '0',
        height: 40,
      },
    },

    li: { ...typos.link, listStyle: 'none' },
    left: {},
    wrapper: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',
      ...(isMobile ? {} : { height: 40 }),
    },
  }

  const childs = isMobile ? (
    <>
      <div>
        {link && <FooterLeftLink isMobile={isMobile} link={link} />}
        {copyright && <li css={styles.li}>{copyright}</li>}
      </div>
      {socials && socials.length > 0 && <FooterSocials socials={socials} />}
    </>
  ) : (
    <>
      {link && <FooterLeftLink link={link} />}
      {copyright && (
        <li key={3} css={styles.li}>
          {copyright}
        </li>
      )}
      {socials && socials.length > 0 && <FooterSocials socials={socials} />}
    </>
  )

  return (
    <footer className={className} css={styles.footer}>
      <Container>
        <div css={styles.wrapper}>{childs}</div>
      </Container>
    </footer>
  )
}

const FooterLeftLink = ({ link, isMobile }) => {
  return (
    <Link
      css={{
        textDecoration: 'underline',
        display: 'block',
        ...(isMobile ? { marginBottom: 10 } : {}),
      }}
      href={link.url}
    >
      {link.label}
    </Link>
  )
}

const FooterSocials = ({ socials }) => {
  const styles = {
    socials: {},
    item: { display: 'block' },
    li: {
      ...typos.link,
      listStyle: 'none',
      float: 'right',
      marginRight: 10,
      '&:first-child': { marginRight: 0 },
    },
  }

  return (
    <nav css={styles.socials}>
      {socials.map((s, i) => (
        <Link key={i} href={s.url} css={styles.li}>
          <Icon css={styles.item} name={s.icon} size="20"></Icon>
        </Link>
      ))}
    </nav>
  )
}
export default Footer

Footer.propTypes = {
  legal: PropTypes.string,
  link: PropTypes.shape({ label: PropTypes.string, url: PropTypes.string }),

  socials: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.string,
      url: PropTypes.string,
    })
  ),
  copyright: PropTypes.string,
}

Footer.defaultProps = {
  link: {},
  legal: '',
  socials: [
    {
      icon: 'adduser',
      url: 'http://wwww.twitter.com',
    },
    {
      icon: 'file',
      url: 'http://wwww.twitter.com',
    },
  ],
  copyright: '',
}
