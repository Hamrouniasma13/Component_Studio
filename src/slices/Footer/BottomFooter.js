import React from 'react'
import PropTypes from 'prop-types'
/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'

import Container from '../../components/Container'
import FooterLang from './FooterLang'
import { textColor, darken } from '../../theme/utils'

function BottomFooter({ langs, lang, langTitle, onLangChange, legal }) {
  const theme = React.useContext(ThemeContext)
  const styles = {
    wrapper: {
      color: textColor(darken(theme.colors.primary, 20), theme),
      backgroundColor: darken(theme.colors.primary, 30),
    },
    legal: {
      fontSize: '12px',
      textAlign: 'center',
      padding: '10px 0',
      [theme.media.tablet]: {
        flex: 1,
        fontSize: '14px',
        '&:not(:first-child)': {
          textAlign: 'right',
        },
      },
      [theme.media.laptop]: {
        textAlign: 'right',
      },
    },
    bottom: {
      label: 'BottomWrapper',
      marginLeft: '-15px',
      marginRight: '-15px',
      [theme.media.tablet]: {
        marginLeft: 0,
        marginRight: 0,
        display: 'flex',
        alignItems: 'center',
        minHeight: '40px',
      },
    },
  }
  return (
    <div css={styles.wrapper}>
      <Container>
        <div css={styles.bottom}>
          {langs && langs.length > 0 ? (
            <FooterLang
              langs={langs}
              lang={lang}
              title={langTitle}
              onLangChange={onLangChange}
            />
          ) : null}
          {legal && <div css={styles.legal}>{legal}</div>}
        </div>
      </Container>
    </div>
  )
}

BottomFooter.propTypes = {
  langs: PropTypes.arrayOf(PropTypes.object),
  legal: PropTypes.string,
  lang: PropTypes.shape({
    ID: PropTypes.string,
    label: PropTypes.string,
  }),
  langTitle: PropTypes.string,
  onLangChange: PropTypes.func,
}

export default BottomFooter
