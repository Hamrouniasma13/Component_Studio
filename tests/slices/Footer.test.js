import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';
import { MemoryRouter } from 'react-router';

import Footer from '../../slices/Footer/Footer';

describe('Component: <Footer />', () => {
  it('render full snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Footer
            langs={[
              {
                ID: 'fr-FR',
                label: 'FranÃ§ais'
              },
              {
                ID: 'en-GB',
                label: 'English'
              }
            ]}
            lang={{
              ID: 'en-GB',
              label: 'English'
            }}
            langTitle={'Languages'}
            links={[
              {
                label: 'More about my first page',
                url: '/my-first-page'
              },
              {
                label: 'Thanks to my friend',
                url: 'https://google.com'
              }
            ]}
            socials={[
              {
                icon: 'facebook',
                url: 'https://facebook.com'
              },
              {
                icon: 'instagram',
                url: 'https://instagram.Com'
              },
              {
                icon: 'twitter',
                url: 'https://twitter.com'
              }
            ]}
            legal={'AXA copycat'}
          />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
