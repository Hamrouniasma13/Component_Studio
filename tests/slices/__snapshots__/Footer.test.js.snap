// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Footer /> render full snapshot 1`] = `
.emotion-17 {
  background-color: #5151e6;
  border-bottom: 1px solid #616AD9;
}

.emotion-16 {
  margin: 0px auto;
  max-width: 100%;
  padding-left: 15px;
  padding-right: 15px;
}

@media only screen and (min-width :768px) {
  .emotion-16 {
    width: 720px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-16 {
    width: 960px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-16 {
    width: 1140px;
  }
}

.emotion-15 {
  margin-left: -15px;
  margin-right: -15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

@media only screen and (min-width :768px) {
  .emotion-15 {
    margin-left: 0;
    margin-right: 0;
  }
}

@media only screen and (min-width :992px) {
  .emotion-15 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

.emotion-4 {
  padding: 20px 15px;
}

.emotion-4 > ul {
  padding-left: 0;
}

@media only screen and (min-width :768px) {
  .emotion-4 {
    padding: 20px 0;
  }
}

.emotion-1 {
  font-size: 14px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  color: #fff;
  padding-right: 10px;
  margin-right: 10px;
}

@media only screen and (min-width :768px) {
  .emotion-1 {
    display: inline-block;
  }
}

.emotion-1:not(:last-child) {
  margin-bottom: 10px;
}

@media only screen and (min-width :768px) {
  .emotion-1:not(:last-child) {
    margin-bottom: 0;
    border-right: 1px solid #e5e5e5;
    padding-right: 10px;
    margin-right: 10px;
  }
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-14 {
  padding: 15px;
  border-bottom: 1px solid #616AD9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-14 > ul {
  padding-left: 0;
}

@media only screen and (min-width :768px) {
  .emotion-14 {
    padding: 13px 0;
    border-bottom: 0;
  }
}

.emotion-7 {
  display: inline-block;
  margin-right: 20px;
  color: #fff;
  margin-bottom: 0px;
}

.emotion-7:last-child {
  margin-right: 0;
}

@media only screen and (min-width :768px) {
  .emotion-7 {
    margin-right: 30px;
  }
}

.emotion-6 {
  display: block;
  line-height: 0;
}

.emotion-5 {
  color: inherit;
  width: 28px;
  height: 28px;
}

.emotion-5 path {
  -webkit-transition: fill 0.3s;
  transition: fill 0.3s;
}

.emotion-5:hover path {
  fill: #e5e5e5;
}

@media only screen and (min-width :768px) {
  .emotion-5 {
    height: 24px;
    width: 24px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-5 {
    height: 20px;
    width: 20px;
  }
}

.emotion-33 {
  background-color: #3133c1;
}

.emotion-31 {
  margin-left: -15px;
  margin-right: -15px;
}

@media only screen and (min-width :768px) {
  .emotion-31 {
    margin-left: 0;
    margin-right: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    min-height: 40px;
  }
}

.emotion-29 {
  border-bottom: 1px solid #616AD9;
  color: white;
  text-transform: uppercase;
  font-size: 14px;
  -webkit-letter-spacing: .6px;
  -moz-letter-spacing: .6px;
  -ms-letter-spacing: .6px;
  letter-spacing: .6px;
  cursor: pointer;
}

@media only screen and (min-width :768px) {
  .emotion-29 {
    min-width: 320px;
    margin: 0;
    border-bottom: 0;
    border-right: 1px solid #616AD9;
    -webkit-transition: background-color 0.2s;
    transition: background-color 0.2s;
    position: relative;
    left: -15px;
  }

  .emotion-29:hover {
    background-color: #262896;
  }
}

.emotion-21 {
  position: relative;
}

.emotion-19 {
  padding-left: 15px;
  height: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-18 {
  color: inherit;
  margin-right: 10px;
  vertical-align: sub;
}

.emotion-20 {
  color: inherit;
  height: 28px;
  width: 28px;
  -webkit-transition: -webkit-transform 0.3s,color 0.3s;
  -webkit-transition: transform 0.3s,color 0.3s;
  transition: transform 0.3s,color 0.3s;
  -webkit-transform: translate(0,-50%) rotate(180deg);
  -ms-transform: translate(0,-50%) rotate(180deg);
  transform: translate(0,-50%) rotate(180deg);
  position: absolute;
  top: 50%;
  right: 15px;
  color: white;
}

.emotion-28 {
  background: white;
  padding: 20px;
  color: #cccccc;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: none;
}

@media only screen and (min-width :768px) {
  .emotion-28 {
    box-shadow: rgba(0,0,0,0.3) 1px 1px 15px 0px;
    position: absolute;
    bottom: 100%;
    left: 0;
    right: 0;
    display: block;
    -webkit-transition: -webkit-transform 0.1s ease 0.1s,opacity 0.3s;
    -webkit-transition: transform 0.1s ease 0.1s,opacity 0.3s;
    transition: transform 0.1s ease 0.1s,opacity 0.3s;
    opacity: 0;
    -webkit-transform: translateY(5px);
    -ms-transform: translateY(5px);
    transform: translateY(5px);
    pointer-events: none;
  }
}

.emotion-22 {
  color: #989898;
  text-transform: capitalize;
}

.emotion-27 {
  border: 1px solid #cccccc;
  margin-top: 10px;
}

.emotion-24 {
  list-style: none;
  padding: 15px;
  color: #989898;
  font-weight: 600;
  border-bottom: 1px solid #cccccc;
  background-color: white;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-24:hover {
  background-color: #e5e5e5;
}

.emotion-24:last-child {
  border-bottom: 0px;
}

.emotion-23 {
  color: inherit;
}

.emotion-25 {
  color: #3133c1;
}

.emotion-30 {
  color: #cccccc;
  font-size: 12px;
  text-align: center;
  padding: 10px 0;
}

@media only screen and (min-width :768px) {
  .emotion-30 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    font-size: 14px;
  }

  .emotion-30:not(:first-child) {
    text-align: right;
  }
}

@media only screen and (min-width :992px) {
  .emotion-30 {
    text-align: right;
  }
}

<footer>
  <div
    className="emotion-17"
  >
    <div
      className="emotion-16"
    >
      <div
        className="emotion-15"
      >
        <div
          className="emotion-4"
        >
          <ul>
            <li
              className="emotion-1"
            >
              <a
                aria-current={null}
                className="emotion-0"
                href="/my-first-page"
                onClick={[Function]}
                style={undefined}
              >
                More about my first page
              </a>
            </li>
            <li
              className="emotion-1"
            >
              <a
                className="emotion-0"
                href="https://google.com"
                onClick={[Function]}
                target="_blank"
              >
                Thanks to my friend
              </a>
            </li>
          </ul>
        </div>
        <div
          className="emotion-14"
        >
          <ul>
            <li
              className="emotion-7"
            >
              <a
                className="emotion-6"
                href="https://facebook.com"
                onClick={[Function]}
                target="_blank"
              >
                <div
                  className="emotion-5"
                  colorindex={undefined}
                  fill="currentColor"
                  size={30}
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "inline-flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <svg
                    fill="currentColor"
                    height={30}
                    style={
                      Object {
                        "display": "inline-block",
                        "verticalAlign": "middle",
                      }
                    }
                    viewBox="0 0 1536 1792"
                    width={30}
                  >
                    <path
                      d="M1248 128q119 0 203.5 84.5t84.5 203.5v960q0 119-84.5 203.5t-203.5 84.5h-188v-595h199l30-232h-229v-148q0-56 23.5-84t91.5-28l122-1v-207q-63-9-178-9-136 0-217.5 80t-81.5 226v171h-200v232h200v595h-532q-119 0-203.5-84.5t-84.5-203.5v-960q0-119 84.5-203.5t203.5-84.5h960z"
                    />
                  </svg>
                </div>
              </a>
            </li>
            <li
              className="emotion-7"
            >
              <a
                className="emotion-6"
                href="https://instagram.Com"
                onClick={[Function]}
                target="_blank"
              >
                <div
                  className="emotion-5"
                  colorindex={undefined}
                  fill="currentColor"
                  size={30}
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "inline-flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <svg
                    fill="currentColor"
                    height={30}
                    style={
                      Object {
                        "display": "inline-block",
                        "verticalAlign": "middle",
                      }
                    }
                    viewBox="0 0 1536 1792"
                    width={30}
                  >
                    <path
                      d="M1024 896q0-106-75-181t-181-75-181 75-75 181 75 181 181 75 181-75 75-181zM1162 896q0 164-115 279t-279 115-279-115-115-279 115-279 279-115 279 115 115 279zM1270 486q0 38-27 65t-65 27-65-27-27-65 27-65 65-27 65 27 27 65zM768 266q-7 0-76.5-0.5t-105.5 0-96.5 3-103 10-71.5 18.5q-50 20-88 58t-58 88q-11 29-18.5 71.5t-10 103-3 96.5 0 105.5 0.5 76.5-0.5 76.5 0 105.5 3 96.5 10 103 18.5 71.5q20 50 58 88t88 58q29 11 71.5 18.5t103 10 96.5 3 105.5 0 76.5-0.5 76.5 0.5 105.5 0 96.5-3 103-10 71.5-18.5q50-20 88-58t58-88q11-29 18.5-71.5t10-103 3-96.5 0-105.5-0.5-76.5 0.5-76.5 0-105.5-3-96.5-10-103-18.5-71.5q-20-50-58-88t-88-58q-29-11-71.5-18.5t-103-10-96.5-3-105.5 0-76.5 0.5zM1536 896q0 229-5 317-10 208-124 322t-322 124q-88 5-317 5t-317-5q-208-10-322-124t-124-322q-5-88-5-317t5-317q10-208 124-322t322-124q88-5 317-5t317 5q208 10 322 124t124 322q5 88 5 317z"
                    />
                  </svg>
                </div>
              </a>
            </li>
            <li
              className="emotion-7"
            >
              <a
                className="emotion-6"
                href="https://twitter.com"
                onClick={[Function]}
                target="_blank"
              >
                <div
                  className="emotion-5"
                  colorindex={undefined}
                  fill="currentColor"
                  size={30}
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "inline-flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <svg
                    fill="currentColor"
                    height={30}
                    style={
                      Object {
                        "display": "inline-block",
                        "verticalAlign": "middle",
                      }
                    }
                    viewBox="0 0 1664 1792"
                    width={30}
                  >
                    <path
                      d="M1620 408q-67 98-162 167 1 14 1 42 0 130-38 259.5t-115.5 248.5-184.5 210.5-258 146-323 54.5q-271 0-496-145 35 4 78 4 225 0 401-138-105-2-188-64.5t-114-159.5q33 5 61 5 43 0 85-11-112-23-185.5-111.5t-73.5-205.5v-4q68 38 146 41-66-44-105-115t-39-154q0-88 44-163 121 149 294.5 238.5t371.5 99.5q-8-38-8-74 0-134 94.5-228.5t228.5-94.5q140 0 236 102 109-21 205-78-37 115-142 178 93-10 186-50z"
                    />
                  </svg>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="emotion-33"
  >
    <div
      className="emotion-16"
    >
      <div
        className="emotion-31"
      >
        <div
          className="emotion-29"
        >
          <div
            className="emotion-21"
            onClick={[Function]}
          >
            <div
              className="emotion-19"
            >
              <div
                className="emotion-18"
                colorindex={undefined}
                fill="currentColor"
                size={20}
                style={
                  Object {
                    "alignItems": "center",
                    "display": "inline-flex",
                    "justifyContent": "center",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height={20}
                  style={
                    Object {
                      "display": "inline-block",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 1536 1792"
                  width={20}
                >
                  <path
                    d="M768 128q209 0 385.5 103t279.5 279.5 103 385.5-103 385.5-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103zM1042 649q-2 1-9.5 9.5t-13.5 9.5q2 0 4.5-5t5-11 3.5-7q6-7 22-15 14-6 52-12 34-8 51 11-2-2 9.5-13t14.5-12q3-2 15-4.5t15-7.5l2-22q-12 1-17.5-7t-6.5-21q0 2-6 8 0-7-4.5-8t-11.5 1-9 1q-10-3-15-7.5t-8-16.5-4-15q-2-5-9.5-11t-9.5-10q-1-2-2.5-5.5t-3-6.5-4-5.5-5.5-2.5-7 5-7.5 10-4.5 5q-3-2-6-1.5t-4.5 1-4.5 3-5 3.5q-3 2-8.5 3t-8.5 2q15-5-1-11-10-4-16-3 9-4 7.5-12t-8.5-14h5q-1-4-8.5-8.5t-17.5-8.5-13-6q-8-5-34-9.5t-33-0.5q-5 6-4.5 10.5t4 14 3.5 12.5q1 6-5.5 13t-6.5 12q0 7 14 15.5t10 21.5q-3 8-16 16t-16 12q-5 8-1.5 18.5t10.5 16.5q2 2 1.5 4t-3.5 4.5-5.5 4-6.5 3.5l-3 2q-11 5-20.5-6t-13.5-26q-7-25-16-30-23-8-29 1-5-13-41-26-25-9-58-4 6-1 0-15-7-15-19-12 3-6 4-17.5t1-13.5q3-13 12-23 1-1 7-8.5t9.5-13.5 0.5-6q35 4 50-11 5-5 11.5-17t10.5-17q9-6 14-5.5t14.5 5.5 14.5 5q14 1 15.5-11t-7.5-20q12 1 3-17-4-7-8-9-12-4-27 5-8 4 2 8-1-1-9.5 10.5t-16.5 17.5-16-5q-1-1-5.5-13.5t-9.5-13.5q-8 0-16 15 3-8-11-15t-24-8q19-12-8-27-7-4-20.5-5t-19.5 4q-5 7-5.5 11.5t5 8 10.5 5.5 11.5 4 8.5 3q14 10 8 14-2 1-8.5 3.5t-11.5 4.5-6 4q-3 4 0 14t-2 14q-5-5-9-17.5t-7-16.5q7 9-25 6l-10-1q-4 0-16 2t-20.5 1-13.5-8q-4-8 0-20 1-4 4-2-4-3-11-9.5t-10-8.5q-46 15-94 41 6 1 12-1 5-2 13-6.5t10-5.5q34-14 42-7l5-5q14 16 20 25-7-4-30-1-20 6-22 12 7 12 5 18-4-3-11.5-10t-14.5-11-15-5q-16 0-22 1-146 80-235 222 7 7 12 8 4 1 5 9t2.5 11 11.5-3q9 8 3 19 1-1 44 27 19 17 21 21 3 11-10 18-1-2-9-9t-9-4q-3 5 0.5 18.5t10.5 12.5q-7 0-9.5 16t-2.5 35.5-1 23.5l2 1q-3 12 5.5 34.5t21.5 19.5q-13 3 20 43 6 8 8 9 3 2 12 7.5t15 10 10 10.5q4 5 10 22.5t14 23.5q-2 6 9.5 20t10.5 23q-1 0-2.5 1t-2.5 1q3 7 15.5 14t15.5 13q1 3 2 10t3 11 8 2q2-20-24-62-15-25-17-29-3-5-5.5-15.5t-4.5-14.5q2 0 6 1.5t8.5 3.5 7.5 4 2 3q-3 7 2 17.5t12 18.5 17 19 12 13q6 6 14 19.5t0 13.5q9 0 20 10.5t17 19.5q5 8 8 26t5 24q2 7 8.5 13.5t12.5 9.5l16 8t13 7q5 2 18.5 10.5t21.5 11.5q10 4 16 4t14.5-2.5 13.5-3.5q15-2 29 15t21 21q36 19 55 11-2 1 0.5 7.5t8 15.5 9 14.5 5.5 8.5q5 6 18 15t18 15q6-4 7-9-3 8 7 20t18 10q14-3 14-32-31 15-49-18 0-1-2.5-5.5t-4-8.5-2.5-8.5 0-7.5 5-3q9 0 10-3.5t-2-12.5-4-13q-1-8-11-20t-12-15q-5 9-16 8t-16-9q0 1-1.5 5.5t-1.5 6.5q-13 0-15-1 1-3 2.5-17.5t3.5-22.5q1-4 5.5-12t7.5-14.5 4-12.5-4.5-9.5-17.5-2.5q-19 1-26 20-1 3-3 10.5t-5 11.5-9 7q-7 3-24 2t-24-5q-13-8-22.5-29t-9.5-37q0-10 2.5-26.5t3-25-5.5-24.5q3-2 9-9.5t10-10.5q2-1 4.5-1.5t4.5 0 4-1.5 3-6q-1-1-4-3-3-3-4-3 7 3 28.5-1.5t27.5 1.5q15 11 22-2 0-1-2.5-9.5t-0.5-13.5q5 27 29 9 3 3 15.5 5t17.5 5q3 2 7 5.5t5.5 4.5 5-0.5 8.5-6.5q10 14 12 24 11 40 19 44 7 3 11 2t4.5-9.5 0-14-1.5-12.5l-1-8v-18l-1-8q-15-3-18.5-12t1.5-18.5 15-18.5q1-1 8-3.5t15.5-6.5 12.5-8q21-19 15-35 7 0 11-9-1 0-5-3t-7.5-5-4.5-2q9-5 2-16 5-3 7.5-11t7.5-10q9 12 21 2 8-8 1-16 5-7 20.5-10.5t18.5-9.5q7 2 8-2t1-12 3-12q4-5 15-9t13-5l17-11q3-4 0-4 18 2 31-11 10-11-6-20 3-6-3-9.5t-15-5.5q3-1 11.5-0.5t10.5-1.5q15-10-7-16-17-5-43 12zM879 1526q206-36 351-189-3-3-12.5-4.5t-12.5-3.5q-18-7-24-8 1-7-2.5-13t-8-9-12.5-8-11-7q-2-2-7-6t-7-5.5-7.5-4.5-8.5-2-10 1l-3 1q-3 1-5.5 2.5t-5.5 3-4 3 0 2.5q-21-17-36-22-5-1-11-5.5t-10.5-7-10-1.5-11.5 7q-5 5-6 15t-2 13q-7-5 0-17.5t2-18.5q-3-6-10.5-4.5t-12 4.5-11.5 8.5-9 6.5-8.5 5.5-8.5 7.5q-3 4-6 12t-5 11q-2-4-11.5-6.5t-9.5-5.5q2 10 4 35t5 38q7 31-12 48-27 25-29 40-4 22 12 26 0 7-8 20.5t-7 21.5q0 6 2 16z"
                  />
                </svg>
              </div>
              <span>
                English
              </span>
            </div>
            <div
              className="emotion-20"
              colorindex={undefined}
              fill="currentColor"
              open={false}
              size={28}
              style={
                Object {
                  "alignItems": "center",
                  "display": "inline-flex",
                  "justifyContent": "center",
                }
              }
            >
              <svg
                fill="currentColor"
                height={28}
                style={
                  Object {
                    "display": "inline-block",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 24 24"
                width={28}
              >
                <path
                  d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
                />
              </svg>
            </div>
          </div>
          <div
            className="emotion-28"
          >
            <div
              className="emotion-22"
            >
              Languages
            </div>
            <div
              className="emotion-27"
            >
              <div
                className="emotion-24"
                onClick={[Function]}
              >
                <div
                  className="emotion-23"
                  selected={false}
                >
                  Français
                </div>
              </div>
              <div
                className="emotion-24"
                onClick={[Function]}
              >
                <div
                  className="emotion-25"
                  selected={true}
                >
                  English
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="emotion-30"
        >
          AXA copycat
        </div>
      </div>
    </div>
  </div>
</footer>
`;
